File: code/seungwookim99/week3/prog_64062.java
class Solution {
  private boolean isPossible(int n, int[] stones, int k) {
      int count = 0;
      for(int i = 0 ; i < stones.length ; i++) {
          if (stones[i] < n){
              if (++count == k) return false;
          }
          else count = 0;
      }
      return true;
  }
  
  public int solution(int[] stones, int k) {
      int answer = 0;
      int left = 1, right = 200000000, mid;
      while (left <= right) {
          mid = (left + right) / 2;
          if (isPossible(mid, stones, k)) {
              left = mid + 1;
              answer = mid;
          } else {
              right = mid - 1;
          }
      }
      return answer;
  }
}

File: code/seungwookim99/week3/prog_64062.py
# Python 풀이
def canPass(num, stones, k):
    count = 0
    for i in range(len(stones)):
        if stones[i] < num:
            count += 1
        else:
            count = 0
        if count == k:
            return False
    return True


def solution(stones, k):
    answer = 0
    right, left = max(stones), min(stones)
    while left <= right:
        mid = (left + right) // 2
        if canPass(mid, stones, k):
            answer = max(answer, mid)
            left = mid + 1
        else:
            right = mid - 1
    return answer
